src/metabase/driver/mongo/util.clj

```diff
@@ -40,12 +44,13 @@
   "Run F with a new connection (bound to `*mongo-connection*`) to DATABASE.
    Don't use this directly; use `with-mongo-connection`."
   [f database]
-  (let [{:keys [dbname host port user pass]
-         :or   {port 27017, pass ""}} (cond
-                                        (string? database)            {:dbname database}
-                                        (:dbname (:details database)) (:details database) ; entire Database obj
-                                        (:dbname database)            database            ; connection details map only
-                                        :else                         (throw (Exception. (str "with-mongo-connection failed: bad connection details:" (:details database)))))
+  ;; The Mongo SSL detail is keyed by :use-ssl because the frontend has accidentally been saving all of the Mongo DBs with {:ssl true}
+  (let [{:keys [dbname host port user pass use-ssl]
+         :or   {port 27017, pass "", use-ssl false}} (cond
```

Comment:
i thought the dictionary was going to have `:ssl` in it instead of `:use-ssl`, or am i misunderstanding how the input makes it to this function?